manager/state/raft/raft.go

```diff
@@ -461,14 +459,13 @@ func (n *Node) Run(ctx context.Context) error {
 			// If we are the only registered member after
 			// restoring from the state, campaign to be the
 			// leader.
-			if !n.restored {
-				// Node ID should be in the progress list to Campaign
-				if len(n.cluster.Members()) <= 1 {
+			if n.campaignWhenAble && n.addedSelf {
+				if len(n.cluster.Members()) == 1 {
```

Comment:
I think this should just get `n.cluster.Members()` and check that sole member is self. Then you don't need `addedSelf` variable.